sdk/core/Azure.Core/src/Shared/AzureEventSource.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+
+namespace Azure.Core.Diagnostics
+{
+    internal abstract class AzureEventSource: EventSource
+    {
+        private static string[] MainEventSourceTraits =
+        {
+            AzureEventSourceListener.TraitName,
+            AzureEventSourceListener.TraitValue
+        };
+
+        protected AzureEventSource(string eventSourceName): base(
+            DeduplicateName(eventSourceName),
+            EventSourceSettings.Default,
+            MainEventSourceTraits
+        )
+        {
+        }
+
+        private static string DeduplicateName(string eventSourceName)
+        {
+            HashSet<string> namesInUse = new();
+            foreach (var source in GetSources())
+            {
+                namesInUse.Add(source.Name);
+            }
+
+            if (!namesInUse.Contains(eventSourceName))
+            {
+                return eventSourceName;
+            }
```

Comment:
```suggestion
            bool nameFound = false;
            foreach (var source in GetSources())
            {
                if (namesInUse.Contains(eventSourceName))
                {
                    nameFound = true;
                    break;
                }
                else
                {
                    namesInUse.Add(source.Name);
                }
            }

            if (!nameFound)
            {
                return eventSourceName;
            }
```

Reply:
Maybe something to not have to build the full set every time.

Reply:
I don't think what you are proposing works. We need to find all existing eventsources `Azure-Core`, `Azure-Core-1`, `Azure-Core-2` to be able to generate `Azure-Core-3` name.